### PR TITLE
Improvements to Stack(Pivot Longer) dialog

### PR DESCRIPTION
**Pull request by Vitalis95**
Partially solves #6868
@rdstern @shadrackkibet, this is for the Drop Missing Values Checkbox. Have a look at it.

### CODE REVIEW

**Comment by Vitalis95** [comment]
@shadrackkibet, have a look at it.

**Comment by lloyddewit** [comment]
@Vitalis95 Please could you resolve @N-thony 's comment above? thanks

**Review by rdstern — DISMISSED** [review]
@Vitalis95 to me this looks fine now.  Let's see what @shadrackkibet thinks?

**Comment by N-thony** [comment]
Push a bit  Name to label to the left, so that it aligns properly with other controls on the left side. Same case with New Data Frame Name label
image

File: instat/dlgStack.vb [hunk 1]
```diff
@@ -281,8 +339,21 @@ Public Class dlgStack
...
+
+    Private Sub SetSingleOrMultipleOptions()
+        If ucrChkStackMultipleSets.Checked Then
+            ucrBase.clsRsyntax.SetBaseRFunction(clsReshapeFunction)
+            ucrReceiverColumnsToBeStack.SetMeAsReceiver()
+        Else
+                ucrBase.clsRsyntax.SetBaseRFunction(clsPivotLongerFunction)
+        End If
+        TestOKEnabled()
+    End Sub
```
Comment (N-thony) [inline instat/dlgStack.vb:355]: Add one line space after line 355

**Comment by Vitalis95** [comment]
@N-thony , I have made the alignments.

**Review by N-thony — APPROVED** [review]

**Review by rdstern — DISMISSED** [review]
Looks good now.

File: instat/dlgStack.vb [hunk 2]
```diff
@@ -19,6 +19,8 @@ Public Class dlgStack
     Private clsUnnestTokensFunction As RFunction
     Private clsPivotLongerFunction As RFunction
     Private clsSelectFunction As RFunction
+    Private clsReshapeFunction As RFunction
+    Private clsSplitColumnsInGroups As RFunction
```
Comment (lloyddewit) [inline instat/dlgStack.vb:23]: Rename to make it clearer that its a function?
```suggestion
    Private clsSplitColumnsFunction As RFunction
```
or a different name if you prefer

File: instat/dlgStack.vb [hunk 3]
```diff
@@ -167,6 +206,16 @@ Public Class dlgStack
         clsSelectFunction.SetRCommand("select")
         clsSelectFunction.SetPackageName("dplyr")
 
+        clsReshapeFunction.SetPackageName("stats")
+        clsReshapeFunction.SetRCommand("reshape")
+        clsReshapeFunction.SetAssignTo(ucrSelectorStack.ucrAvailableDataFrames.cboAvailableDataFrames.Text & "_stacked", strTempDataframe:=ucrSelectorStack.ucrAvailableDataFrames.cboAvailableDataFrames.Text & "_stacked")
+        clsReshapeFunction.AddParameter("direction", Chr(34) & "long" & Chr(34))
```
Comment (lloyddewit) [inline instat/dlgStack.vb:212]: Would it be safer and more readable to explicitly set `iPosition:=0'?

File: instat/dlgStack.vb [hunk 4]
```diff
@@ -202,7 +258,7 @@ Public Class dlgStack
                 ucrBase.OKEnabled(True)
             End If
         Else
-            If ucrReceiverColumnsToBeStack.IsEmpty OrElse Not ucrSaveNewDataName.IsComplete OrElse ucrInputNamesTo.IsEmpty OrElse ucrInputValuesTo.IsEmpty OrElse (ucrReceiverColumnsToCarry.IsEmpty AndAlso ucrChkCarryColumns.Checked) Then
+            If ucrReceiverColumnsToBeStack.IsEmpty OrElse Not ucrSaveNewDataName.IsComplete OrElse ucrInputNamesTo.IsEmpty OrElse ucrInputValuesTo.IsEmpty OrElse ucrFactorInto.IsEmpty OrElse (ucrReceiverColumnsToCarry.IsEmpty AndAlso ucrChkCarryColumns.Checked) Then
```
Comment (lloyddewit) [inline instat/dlgStack.vb:261]: add line breaks so that each line is max 100 columns

File: instat/dlgStack.vb [hunk 5]
```diff
@@ -281,8 +339,22 @@ Public Class dlgStack
...
-    ucrSaveNewDataName.ControlContentsChanged, ucrInputOutput.ControlContentsChanged, ucrReceiverTextColumn.ControlContentsChanged, ucrInputToken.ControlContentsChanged,
-    ucrPnlStack.ControlContentsChanged, ucrInputFormat.ControlContentsChanged, ucrInputPattern.ControlContentsChanged, ucrChkCarryColumns.ControlContentsChanged, ucrReceiverColumnsToCarry.ControlContentsChanged
+    ucrSaveNewDataName.ControlContentsChanged, ucrInputOutput.ControlContentsChanged, ucrReceiverTextColumn.ControlContentsChanged, ucrInputToken.ControlContentsChanged, ucrInputDropPrefix.ControlContentsChanged, ucrFactorInto.ControlContentsChanged,
+    ucrPnlStack.ControlContentsChanged, ucrInputFormat.ControlContentsChanged, ucrInputPattern.ControlContentsChanged, ucrChkCarryColumns.ControlContentsChanged, ucrReceiverColumnsToCarry.ControlContentsChanged, ucrChkDropMissingValues.ControlContentsChanged, ucrChkDropPrefix.ControlContentsChanged, ucrChkStackMultipleSets.ControlContentsChanged
+        TestOKEnabled()
+    End Sub
+
+    Private Sub SetSingleOrMultipleOptions()
```
Comment (lloyddewit) [inline instat/dlgStack.vb:347]: This sub is only called from one location (line 358).
Delete this sub and move its contents to line 358?

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Vitalis95 Thanks, this looks good, just a few suggestions

**Comment by Vitalis95** [comment]
@lloyddewit  I have made the changes. Thanks

**Review by lloyddewit — APPROVED** [review]
@Vitalis95 thanks, looks good

**Comment by lloyddewit** [comment]
@rdstern there's been some small changes since your last approval, please could you retest? thanks

**Review by rdstern — APPROVED** [review]
Great!

**Pull request merged**